src/gist.c

```diff
@@ -1672,6 +1677,127 @@ fallbackSplit(Box3D *boxes, OffsetNumber maxoff, GIST_SPLITVEC *v)
 	v->spl_ldatum_exists = v->spl_rdatum_exists = false;
 }
 
+
+Datum
+g_spoint_distance(PG_FUNCTION_ARGS)
+{
+	GISTENTRY  *entry = (GISTENTRY *) PG_GETARG_POINTER(0);
+	StrategyNumber strategy = (StrategyNumber) PG_GETARG_UINT16(2);
+	Box3D* box = (Box3D *) DatumGetPointer(entry->key);
+	double		retval;
+	SPoint	   *point = (SPoint *) PG_GETARG_POINTER(1);
+	Vector3D 	v_point, v_low, v_high;
+
+	switch (strategy)
+	{
+		case 17:
+			// Prepare data for calculation
+			spoint_vector3d(&v_point, point);
+			v_low.x  = (double)box->low.coord[0]  / MAXCVALUE;
+			v_low.y  = (double)box->low.coord[1]  / MAXCVALUE;
+			v_low.z  = (double)box->low.coord[2]  / MAXCVALUE;
+			v_high.x = (double)box->high.coord[0]  / MAXCVALUE;
+			v_high.y = (double)box->high.coord[1]  / MAXCVALUE;
+			v_high.z = (double)box->high.coord[2]  / MAXCVALUE;
+			//  a box splits space into 27 subspaces (6+12+8+1) with different distance calculation
+			if(v_point.x < v_low.x) { 
+				if(v_point.y < v_low.y) { 
+					if(v_point.z < v_low.z) { 
+						retval = distance_vector_point_3d (&v_point, v_low.x, v_low.y, v_low.z); //point2point distance
```

Comment:
Change `distance_vector_point_3d (` to `distance_vector_point_3d(` throughout this function in order to match the coding style of pgsphere.